### ISSUE TITLE
Solve NLI in a better way for sentence representations

### ISSUE THREAD

**Issue by s1m0000n** [s1m0000n/multitask_transformers_ru#6]
Use Triplet Loss or Softmax with separate sentence encoding as in Sentence-BERT to have better sentence representations

**Comment by s1m0000n** [comment]
Explanations about Bi/Cross-Encoders: https://www.sbert.net/examples/applications/cross-encoder/README.html
Implementation of Bi-Encoder:
 - Independent encoding: https://github.com/UKPLab/sentence-transformers/blob/master/sentence_transformers/SentenceTransformer.py
 - Training, higher level(attention to inputs!) https://github.com/UKPLab/sentence-transformers/blob/master/examples/training/nli/training_nli_v2.py
 - One of the Loss classes, for example SoftmaxLoss, making a forward pass! Not in SentenceTransformer.fit: https://github.com/UKPLab/sentence-transformers/blob/master/sentence_transformers/losses/SoftmaxLoss.py
